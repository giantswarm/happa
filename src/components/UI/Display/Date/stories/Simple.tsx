import { Story } from '@storybook/react';
import React from 'react';

import Date from '..';

export const Simple: Story<React.ComponentPropsWithoutRef<typeof Date>> = (
  args
) => {
  return (
    <div>
      <Date {...args} />
    </div>
  );
};

Simple.args = {};

Simple.argTypes = {
  value: {
    control: {
      type: 'date',
    },
    defaultValue: '2021-01-10',
    name: 'date (local time zone)',
  },
};
