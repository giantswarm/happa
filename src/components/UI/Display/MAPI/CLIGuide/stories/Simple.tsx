import { Story } from '@storybook/react';
import { Box, Text } from 'grommet';
import React from 'react';
import { CodeBlock, Prompt } from 'UI/Display/Documentation/CodeBlock';

import CLIGuide from '..';

export const Simple: Story<React.ComponentPropsWithoutRef<typeof CLIGuide>> = (
  args
) => {
  return <CLIGuide {...args} />;
};

Simple.args = {
  title: 'Get some data from the somewhere',
  children: (
    <Box direction='column' gap='small'>
      <Box direction='column' gap='small'>
        <Text>1. Make sure you are a boss</Text>
        <CodeBlock>
          <Prompt>some command --some-flag</Prompt>
        </CodeBlock>
      </Box>
      <Box direction='column' gap='small'>
        <Text>2. Maybe check again if you are still a boss</Text>
        <CodeBlock>
          <Prompt>some-other-cli checkboss</Prompt>
        </CodeBlock>
      </Box>
    </Box>
  ),
};

Simple.argTypes = {
  title: { control: { type: 'text' } },
};
