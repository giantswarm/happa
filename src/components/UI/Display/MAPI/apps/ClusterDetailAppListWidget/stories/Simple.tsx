import { Story } from '@storybook/react';
import { Box, Heading, Text } from 'grommet';
import React, { ComponentPropsWithoutRef } from 'react';

import ClusterDetailAppListWidget from '..';

export const Simple: Story<
  ComponentPropsWithoutRef<typeof ClusterDetailAppListWidget>
> = (args) => {
  return (
    <ClusterDetailAppListWidget {...args}>
      <Text>Hi everyone. This is a widget that you can customize</Text>
      <Box direction='row' pad='small' gap='small' margin={{ top: 'medium' }}>
        <Box align='center'>
          <Heading level={3} margin='none'>
            50
          </Heading>
          <Text>shades of happa</Text>
        </Box>
        <Box align='center'>
          <Heading level={3} margin='none'>
            50
          </Heading>
          <Text>shades of happa</Text>
        </Box>
        <Box align='center'>
          <Heading level={3} margin='none'>
            50
          </Heading>
          <Text>shades of happa</Text>
        </Box>
      </Box>
    </ClusterDetailAppListWidget>
  );
};

Simple.args = {
  title: 'Some widget',
};

Simple.argTypes = {
  title: { control: { type: 'text' } },
};
