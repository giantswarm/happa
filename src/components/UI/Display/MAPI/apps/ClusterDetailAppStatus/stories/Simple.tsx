import { Story } from '@storybook/react';
import { Box } from 'grommet';
import React, { ComponentPropsWithoutRef } from 'react';

import ClusterDetailAppStatus from '..';

export const Simple: Story<
  ComponentPropsWithoutRef<typeof ClusterDetailAppStatus>
> = (args) => {
  return (
    <Box gap='medium' align='flex-start'>
      <ClusterDetailAppStatus {...args} />
    </Box>
  );
};

Simple.args = {
  status: 'some-status',
};

Simple.argTypes = {
  status: { control: { type: 'text' } },
};
