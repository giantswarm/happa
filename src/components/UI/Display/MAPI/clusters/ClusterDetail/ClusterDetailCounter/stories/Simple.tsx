import { Story } from '@storybook/react';
import React, { ComponentPropsWithoutRef } from 'react';

import ClusterDetailCounter from '..';

export const Simple: Story<
  ComponentPropsWithoutRef<typeof ClusterDetailCounter>
> = (args) => {
  return <ClusterDetailCounter {...args} />;
};

Simple.args = {
  label: 'dog',
  value: 1,
};

Simple.argTypes = {
  label: { control: { type: 'text' } },
  value: { control: { type: 'number' } },
};
