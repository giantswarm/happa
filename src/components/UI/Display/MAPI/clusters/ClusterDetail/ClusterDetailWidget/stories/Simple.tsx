import { Story } from '@storybook/react';
import { Box, Heading, Text } from 'grommet';
import React, { ComponentPropsWithoutRef } from 'react';

import ClusterDetailWidget from '..';

export const Simple: Story<
  ComponentPropsWithoutRef<typeof ClusterDetailWidget>
> = (args) => {
  return (
    <ClusterDetailWidget {...args}>
      <Text>Hi everyone. This is a widget that you can customize</Text>
      <Box direction='row' pad='small' gap='small' margin={{ top: 'medium' }}>
        <Box align='center'>
          <Heading level={3} margin='none'>
            50
          </Heading>
          <Text>shades of happa</Text>
        </Box>
        <Box align='center'>
          <Heading level={3} margin='none'>
            50
          </Heading>
          <Text>shades of happa</Text>
        </Box>
        <Box align='center'>
          <Heading level={3} margin='none'>
            50
          </Heading>
          <Text>shades of happa</Text>
        </Box>
      </Box>
    </ClusterDetailWidget>
  );
};

Simple.args = {
  title: 'Some widget',
};

Simple.argTypes = {
  title: { control: { type: 'text' } },
};
