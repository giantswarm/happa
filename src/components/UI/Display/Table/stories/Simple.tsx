import { Story } from '@storybook/react';
import React, { ComponentPropsWithoutRef } from 'react';

import {
  Table,
  TableBody,
  TableCell,
  TableFooter,
  TableHeader,
  TableRow,
} from '..';

export const Simple: Story<ComponentPropsWithoutRef<typeof Table>> = (args) => {
  return (
    <Table {...args}>
      <TableHeader>
        <TableRow>
          <TableCell>Name</TableCell>
          <TableCell>Location</TableCell>
          <TableCell>Amount</TableCell>
        </TableRow>
      </TableHeader>
      <TableBody>
        <TableRow>
          <TableCell>Food</TableCell>
          <TableCell>Amazon</TableCell>
          <TableCell>$25,36</TableCell>
        </TableRow>
        <TableRow>
          <TableCell>Laptop</TableCell>
          <TableCell>Apple</TableCell>
          <TableCell>$1683,00</TableCell>
        </TableRow>
        <TableRow>
          <TableCell>Phone</TableCell>
          <TableCell>Apple</TableCell>
          <TableCell>$1037,73</TableCell>
        </TableRow>
      </TableBody>
      <TableFooter>
        <TableRow>
          <TableCell />
          <TableCell />
          <TableCell>$2746,09</TableCell>
        </TableRow>
      </TableFooter>
    </Table>
  );
};

Simple.args = {
  caption: 'Expenses',
};

Simple.argTypes = {};
