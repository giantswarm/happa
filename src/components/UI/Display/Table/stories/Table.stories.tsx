import { Meta } from '@storybook/react';

import { Table } from '..';

export { Simple } from './Simple';

export default {
  title: 'Display/Table',
  component: Table,
} as Meta;
