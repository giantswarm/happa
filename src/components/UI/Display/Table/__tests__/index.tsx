import { renderWithTheme } from 'test/renderUtils';

import { Simple } from '../stories/Simple';

describe('Table', () => {
  it('renders a simple table', () => {
    const { container } = renderWithTheme(Simple, {
      caption: 'Expenses',
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
