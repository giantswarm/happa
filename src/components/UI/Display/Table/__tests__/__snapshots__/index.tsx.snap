// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders a simple table 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 giQQcM"
>
  <div
    class="ThemeProvider__AppWrapper-sc-12xd2cf-0 kuNpSX"
  >
    <table
      class="StyledTable-sc-1m3u5g-6 ixrRkt"
    >
      <caption
        class="StyledTable__StyledTableDataCaption-sc-1m3u5g-1 bSDold"
      >
        Expenses
      </caption>
      <thead
        class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
      >
        <tr
          class="StyledTable__StyledTableRow-sc-1m3u5g-2"
        >
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 kxsoLa"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 hpSXsQ"
            >
              Name
            </div>
          </td>
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 kxsoLa"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 hpSXsQ"
            >
              Location
            </div>
          </td>
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 kxsoLa"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 hpSXsQ"
            >
              Amount
            </div>
          </td>
        </tr>
      </thead>
      <tbody
        class="StyledTable__StyledTableBody-sc-1m3u5g-3"
      >
        <tr
          class="StyledTable__StyledTableRow-sc-1m3u5g-2"
        >
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 hYDwRe"
          >
            Food
          </td>
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 hYDwRe"
          >
            Amazon
          </td>
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 hYDwRe"
          >
            $25,36
          </td>
        </tr>
        <tr
          class="StyledTable__StyledTableRow-sc-1m3u5g-2"
        >
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 hYDwRe"
          >
            Laptop
          </td>
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 hYDwRe"
          >
            Apple
          </td>
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 hYDwRe"
          >
            $1683,00
          </td>
        </tr>
        <tr
          class="StyledTable__StyledTableRow-sc-1m3u5g-2"
        >
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 hYDwRe"
          >
            Phone
          </td>
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 hYDwRe"
          >
            Apple
          </td>
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 hYDwRe"
          >
            $1037,73
          </td>
        </tr>
      </tbody>
      <tfoot
        class="StyledTable__StyledTableFooter-sc-1m3u5g-5"
      >
        <tr
          class="StyledTable__StyledTableRow-sc-1m3u5g-2"
        >
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 dbvYQD"
          />
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 dbvYQD"
          />
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 dbvYQD"
          >
            $2746,09
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;
